### ISSUE TITLE
Clean up legacy test configurations

### ISSUE THREAD

**Issue by natebosch** [dart-lang/sdk#37318]
We should no longer need:

- [ ] preview-dart-2 argument or any tests against no-preview-dart-2
- [ ] Compiler.appJit, replaced with Compiler.appJitk
- [ ] Compiler.precompiled, replaced with Compiler.dartkp
- [ ] checked argument or any tests against unchecked

We _do_ still need host-checked. It would be best to rename to host-asserts-enabled but it may not be worth changing.

**Comment by mkustermann** [comment]
Regarding `we no longer need: checked argument`: The `--checked` flag not only added type checks (which are not needed anymore), but also assertions. We want to continue checking tests with assertions enabled. Some time ago I've filed https://github.com/dart-lang/sdk/issues/31739 for this.

@athomas Do we already have bots running with assertions enabled?

**Comment by athomas** [comment]
There's this one https://ci.chromium.org/p/dart/builders/luci.dart.ci.sandbox/vm-kernel-checked-linux-release-x64 (it uses the --enable-asserts flag).

**Comment by natebosch** [comment]
Right, I'd expect we should be using the [`--enable-asserts` flag](https://github.com/dart-lang/sdk/blob/37a25d9100e802b4804f023fe540e99d6bf2c15f/pkg/smith/lib/configuration.dart#L255)

We only use `checked` in [one place in test_matrix.json](https://github.com/dart-lang/sdk/blob/5b113ee1936b87e93540e95c5b6ce87621573923/tools/bots/test_matrix.json#L359). My assumption was that I should be able to replace that with `enable-asserts: true` which is used more.

**Comment by athomas** [comment]
Interestingly, DDC's compiler configuration doesn't seem to have any code to wire up the checked or enable-asserts flags. I guess the intention here is to pass --enable-asserts to DDC which would be a no-op because assertions are enabled on DDC by default.

**Comment by natebosch** [comment]
Bob handled `--preview-dart-2` in https://github.com/dart-lang/sdk/commit/89ad636b584dacd7b44c7d6f84a6d37e22c6e9c0
